autogluon/task/tabular_prediction/dataset.py

```diff
@@ -30,7 +30,10 @@ class TabularDataset(pd.DataFrame):
     subsample : int (optional)
         If specified = k, we only keep first k rows of the provided dataset.
     name : str (optional)
-         Optional name to assign to dataset (has no effect beyond being accessible via `TabularDataset.name`).
+        Optional name to assign to dataset (has no effect beyond being accessible via `TabularDataset.name`).
+    copy : bool (optional, default=False)
+        If True and `df` is passed, then `df` will be deep copied, resulting in two instances of the provided data in memory.
```

Comment:
Should we add advice where this copy might be useful?  Eg: 
"Set `copy=True` if you don't want AutoGluon to modify your DataFrame during it's training process."

Reply:
Added